### DISCUSSION TITLE
.DMI image format support?

### DISCUSSION THREAD

**Discussion by S34NW** [community/community#72864] [Other Feature Feedback, Questions, & Ideas]
### Select Topic Area

Question

### Body

How would one go about requesting GitHub to add support for viewing contents of .DMI images, a file type commonly used in DM language projects such as SS13? https://github.com/topics/ss13

DMI files are essentially PNG files but with some extra header data, they can be used as a PNG file natively when opened in an image viewer/editor. I'd imagine adding support for this would be straightforward as you would essentially just need to display DMI images as if they were PNG images.

It's a bit niche so I get if they won't want to focus dev time on something like this, but I figured it was worth asking.

**Comment by Naman-B-Parlecha** [comment]
To request GitHub to add support for viewing contents of .DMI images, you can follow these steps:

1. **Visit GitHub's Feedback Page:**
   Go to GitHub's feedback and suggestions page at [https://support.github.com/contact/feedback](https://support.github.com/contact/feedback).

2. **Select "Feature Request":**
   In the "How can we improve GitHub?" section, choose the "Feature request" category.

3. **Provide Details:**
   Clearly explain your request, including the following information:
   - Describe what .DMI files are and their relevance to the community, especially within SS13 projects.
   - Explain that .DMI files are similar to PNG files with additional header data.
   - Suggest that GitHub should treat .DMI files like PNG files for viewing.
   - Mention how this feature could benefit users in the SS13 community.

4. **Attach Examples or Context:**
   If possible, attach examples of .DMI files and how they are commonly used in SS13 projects. This can help GitHub understand the context of your request.

5. **Submit Your Request:**
   Once you've filled out the form with your request and provided all relevant information, submit it.

While it's true that adding support for niche file formats might not be a top priority for GitHub, it's worth making the request, as they do consider user feedback when planning updates and improvements. Additionally, community support and interest in the feature can make a difference. You can also encourage others in the SS13 community to voice their support for this feature request on GitHub's platform.

**Comment by oussamaef06** [comment]
Requesting GitHub to add support for viewing specific file types is not a common practice, and it might not be within their scope of development priorities. However, you can try the following steps if you want to propose this feature:

Create a Detailed Proposal:

Start by creating a detailed proposal for this feature request. Outline what .DMI files are, how they are similar to PNG files, and why adding support for them could benefit the community.
Check for Existing Requests:

Before creating a new feature request, check the GitHub repository or the GitHub Community to see if there are existing feature requests or discussions related to adding support for .DMI files.
Choose the Appropriate Repository:

Identify the appropriate GitHub repository where this feature should be implemented. If it's related to the rendering of image files, it might be in a repository that deals with GitHub's user interface or a specific project management tool.
Open an Issue:

Create a new issue in the selected repository. In the issue, provide a clear and concise description of what you're requesting. Include your proposal, details about .DMI files, and their relevance to SS13 or other projects.
Explain Benefits:

Explain the benefits of adding support for .DMI files, such as improving the user experience for SS13 developers and users. Highlight any potential positive impact on the open-source community.
Be Respectful and Patient:

Remember to be respectful and patient when interacting with the GitHub community. Be open to feedback and discussion.
Engage the Community:

Share the issue with the SS13 community and others who might be interested. Encourage them to add their thoughts and support to the issue. The more interest and support it gathers, the more likely it is to be considered.
Follow Up:

Periodically check for updates on the issue, respond to comments, and clarify any questions that might arise. GitHub's development teams may ask for more details or discuss the proposal.
Contribute Code (Optional):

If you have the technical skills, you can also consider contributing the code yourself. Open-source projects often welcome contributions from the community.
Keep in mind that while your proposal may be valid, GitHub's development priorities are influenced by a variety of factors, including the needs and interests of a broad user base. Therefore, there's no guarantee that your feature request will be implemented, but it's worth trying, especially if there is a strong case for its value to a niche community.
